recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,212 @@
+        "enable_discovery": True,
+        "enable_subscriptions_alarms_conditions": False,
+        "enable_discovery_multicast": False,
+        "enable_parsing": True,
+        "enable_da": True,
+        "enable_micro_emb_dev_profile": False,
+        "enable_websocket_server": False,
+        "enable_query": False,
+        "enable_immutable_nodes": False,
+        "enable_experiemental_historizing": False,
+        "force_cpp": False,
+        "enable_pubsub": False,
+        "enable_pubsub_eth_uadp": False,
+        "enable_pubsub_eth_uadp_xdp": False,
+        "enable_pubsub_deltaframes": False,
+        "enable_pubsub_informationmodel": False,
+        "enable_pubsub_informationmodel_methods": False,
+        "enable_pubsub_custom_publish_handling": False,
+        "enable_pubsub_eth_uadp_etf": False,
+        "enable_json_encoding": False,
+        "enable_pubsub_mqtt": False,
+        "enable_statuscode_descriptions": True,
+        "enable_typedescription": True,
+        "enable_nodeset_compiler_descriptions": True,
+        "enable_malloc_singleton": False,
+        "msvc_force_static_crt": True,
+        "enable_discovery_semaphore": True,
+        "debug_dump_pkgs": False,
+        "enable_hardening": True
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.tls == "openssl":
+            self.requires("openssl/1.1.1g")
+
+        if self.options.tls == "mbedtls":
+            self.requires("mbedtls/2.16.3-apache")
+
+        if self.options.enable_websocket_server:
+            raise ConanInvalidConfiguration("libwebsocket is not (yet) available on CCI")
+            self.requires("libwebsockets/x.y.z")
+
+
+    def source(self):
+        archive_name = self.name + "-" + self.version
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(archive_name, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["UA_ARCHITECTURE"] = "win32" if self.settings.os == "Windows" else "posix"
+
+        if self.options.tls == "openssl":
+            self._cmake.definitions["UA_ENABLE_ENCRYPTION_OPENSSL"] = True
+
+        if self.options.tls == "mbedtls":
+            self._cmake.definitions["UA_ENABLE_ENCRYPTION_MBEDTLS"] = True
+
+        if self.options.multithread == "no":
+            self._cmake.definitions["UA_MULTITHREADING"] = "0"
+
+        if self.options.multithread == "thread-safe":
+            self._cmake.definitions["UA_MULTITHREADING"] = "100"
+
+        if self.options.multithread == "internal-threads":
+            self._cmake.definitions["UA_MULTITHREADING"] = "200"
```

Comment:
Shouldn't it be possible to override these magic values using options?
Something like
```
options =  ... "multithread": "ALL",
default_options = ... "multithread": "False"

def _ua_multithreaded(self):
    if self.options.multithread in ("False", "thread-safe", "internal-threads"):
        return {
            "no": "0",
            "thread-safe": "100",
            "internal-threads": "200",
        }[str(self.options.multithread)]
    else:
        try:
            return str(int(self.options.multithread))
        except ValueError:
            return None

def configure(self):
    if self._ua_multithreaded() is None:
        raise ConanException("Invalid 'multithread' option")

def build(self):
    self._cmake.definitions["UA_MULTITHREADING"] = self._ua_multithreaded()

def package_id(self):
        self.info.options.multithreaded = self._ua_multithreading
```

Unless I'm understanding it wrong.

Reply:
Actually I need to double check if the numbers are not required. In cmakelists there is magic numbers but I have not yet checked in the entire code base.
I pushed this immature recipe for my question about python...
You are right that dict/switch is better than elses...

Reply:
> There is a dependency on python in the CMakeLists though find_package(PythonInterp). Is it possible to get the conans python interpreter, or have a build_requirement on python ?

I don't know of any official convention.

What I did until now is assume `python` is available on a build system.
Nevertheless, you cannot assume that `sys.executable` is the name of a python interpreter because conan is available as an installer using [pyinstaller](https://pyinstaller.readthedocs.io/en/v3.3.1/runtime-information.html).

What you can do is add a [cpython build requirement](https://github.com/conan-io/conan-center-index/pull/1510) if `tools.which("python") or tools.which("python3")` returns `False`.

Reply:
Done